haskell/private/cabal_wrapper.py.tpl

```diff
@@ -53,16 +67,16 @@ def run(cmd, *args, **kwargs):
 def find_exe(exe):
     if os.path.isfile(exe):
         path = os.path.abspath(exe)
-    elif "%{is_windows}" == "True" and os.path.isfile(exe + ".exe"):
+    elif is_windows and os.path.isfile(exe + ".exe"):
         path = os.path.abspath(exe + ".exe")
     else:
         r = bazel_runfiles.Create()
-        path = r.Rlocation("%{workspace}/" + exe)
-        if not os.path.isfile(path) and "%{is_windows}" == "True":
-            path = r.Rlocation("%{workspace}/" + exe + ".exe")
+        path = r.Rlocation(toolchain_info["workspace"] + "/" + exe)
+        if not os.path.isfile(path) and is_windows:
+            path = r.Rlocation(toolchain_info["workspace"] + "/" + exe + ".exe")
```

Comment:
`r.Rlocation` is for discovering runfiles. However, the tools are now no longer runfiles of the cabal wrapper, but instead parameters and build inputs during the cabal build actions. I.e. the paths provided in the parameters should valid relative to the execroot as is. We'll need to make them absolute, since cabal wrapper later changes directory into the directory containing the cabal file.

It's possible that not all of the required hs and cc toolchain components are inputs to the cabal build action yet. You can use `bazel aquery` on some cabal target and inspect the inputs to the `HaskellLibrary|Binary` action.